lua/gatekeeper_config.lua

```diff
@@ -1,5 +1,6 @@
 -- TODO Add configuration for other functional blocks.
 local block_names = {
+	-- The net configuration must come first.
 	"net",
```

Comment:
Remove the comment and `"net"` from this table.

In `gatekeeper_init()`, just before the loop, add the following lines:
```
local net = require("net")
local net_conf = net.setup_block()
if ~net_conf then return -1 end
```

Function `net.setup_block()` in `lua/net.lua` must be changed to return either `net_conf` or `nil`.

Finally, the loop in `gatekeeper_init()` can pass `net_conf` as the parameter of `block.setup_block()`.

The whole change is smaller than its description, and it will structurally make sure that `net_conf` is inilialized first.